ui/packages/consul-ui/app/components/consul/discovery-chain/utils.js

```diff
@@ -124,7 +125,7 @@ export const getResolvers = function(
         const resolver = findResolver(resolvers, target.Service, nspace, partition, dc);
         // and add the redirect as a child, redirects are always children
         const child = {
-          Redirect: true,
+          Redirect: alternate.Type,
```

Comment:
Instead of just knowing if it's a "redirect" or not, we also need to know the type. Instead of adding further properties, we just rely on that fact that if the "redirect" type isn't set, then its not a "redirect". There's a larger question to be answered on what a redirect actually is in the UI, but thats another longer term thing.